[mlir] add a fluent API to GreedyRewriterConfig

ftynse opened the pull request:
This is similar to other configuration objects used across MLIR.

joker-eph commented on mlir/include/mlir/Transforms/GreedyPatternRewriteDriver.h:
```diff
@@ -50,24 +50,41 @@ class GreedyRewriteConfig {
+  GreedyRewriteConfig &setUseTopDownTraversal(bool use = true) {
+    useTopDownTraversal = use;
+    return *this;
+  }
```
I'm all for encapsulation, but if we're adding methods, can we make all the data members private and turn this into a class?

joker-eph replied:
You marked it resolved but I don't see he change in this class?

ftynse replied:
Yeah, pushed the wrong commit...

ftynse committed:
[mlir] add a fluent API to GreedyRewriterConfig

This is similar to other configuration objects used across MLIR.

Rename some fields to better reflect that they are no longer booleans.

kazutakahirata commented:
@ftynse I've reverted this PR as it breaks the release build.  I'm happy to try your revised patch.  Thanks!
